controllers/storageautoscaler/storageautoscaler_controller.go

```diff
@@ -67,6 +67,30 @@ func (r *StorageAutoscalerReconciler) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
+	// if more than one storage autoscaler is present in the cluster with same device class and same storage cluster
+	// return error
+	storageAutoScalerList := &ocsv1.StorageAutoScalerList{}
+	err = r.List(ctx, storageAutoScalerList,
+		client.InNamespace(request.Namespace),
+		client.MatchingFields{"spec.storageCluster.name": storageAutoScaler.Spec.StorageCluster.Name, "spec.deviceClass": storageAutoScaler.Spec.DeviceClass},
+	)
+	if err != nil {
+		r.Log.Error(err, "failed to list storage autoscaler")
+		return reconcile.Result{}, err
+	}
+
+	if len(storageAutoScalerList.Items) > 1 {
+		err = fmt.Errorf("more than one storage autoscaler present with same device class and same storage cluster")
+		r.Log.Error(err, "duplicate cr detected", "device class", storageAutoScaler.Spec.DeviceClass, "storage cluster", storageAutoScaler.Spec.StorageCluster.Name)
+
+		// update the status
+		originalStorageAutoScaler := storageAutoScaler.DeepCopy()
+		duplicateCrsDetected := true
+		storageAutoScaler.Status.CrsWithDuplicateSpecDetected = &duplicateCrsDetected
+		err := r.updateStatus(ctx, storageAutoScaler, originalStorageAutoScaler)
+		return reconcile.Result{}, err
```

Comment:
Which error should be returned , `error` in line#83 or the update status `error` in line#90?